.github/workflows/check-by-name.yml

```diff
@@ -51,4 +61,22 @@ jobs:
           # Passing --max-jobs 0 makes sure that we won't build anything
           nix-build "$nixpkgs" -A tests.nixpkgs-check-by-name --max-jobs 0
       - name: Running nixpkgs-check-by-name
-        run: result/bin/nixpkgs-check-by-name .
+        run: |
+          check() {
+            git checkout -q "$1"
+            result/bin/nixpkgs-check-by-name .
+          }
+
+          echo "Checking whether the check succeeds on the base branch $GITHUB_BASE_REF"
+          if check "$base_sha"; then
+            echo "The check succeeds on the base branch $GITHUB_BASE_REF, now checking the PR"
+            check "$merged_sha"
+          else
+            echo "The check fails on the base branch $GITHUB_BASE_REF, now checking the PR"
+            if check "$merged_sha"; then
+              echo "The check succeeds on the PR, so this fixes the base branch $GITHUB_BASE_REF, nicely done!"
+            else
+              echo "The check also fails on the PR, unknown if only this PRs changes would satisfy the check, the base branch $GITHUB_BASE_REF needs to be fixed"
+              false
+            fi
+          fi
```

Comment:
Perhaps nicer to just present the information succinctly first?

[FAIL] this PR
[FAIL] master
Looks like master is broken. master should be fixed first.

Reply:
what about adding a nice job summary https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/